Fixed #27891 -- Added PasswordResetConfirmView.post_reset_login_backend attribute.

camilonova opened the pull request:
https://code.djangoproject.com/ticket/27891#ticket

timgraham commented:
I pushed a commit with some cosmetic edits. The test needs some enhancement so that it works as a regression test if `auth_login(self.request, user, self.post_reset_login_backend)` is changed back to `auth_login(self.request, user)`,

timgraham commented:
Is that really a regression test? I think that would pass before and after this patch. I think the regression test would be to use multiple auth backends in the original test case and verify that the one selected in the URLconf is used.

camilonova commented:
The test should not pass before this patch. That was exactly the error I was getting before working on this bug.

I'll make right now another test as you suggest.

timgraham commented:
`test_confirm_login_post_reset_custom_backend_regresion` doesn't use a custom backend, so that error happens before and after the patch. By the way, you can rebase and squash commits when updating. Thanks.

camilonova commented:
Tim, you were right, I just removed the old test and updated the old one to check the backend used.

Please tell me if you think is fine and I'll rebase and squash.

Thanks.

timgraham commented:
Yes, that's what I expected. I think it would be slightly clearer if the urlpattern used the second auth backend (AllowAllUsersModelBackend) instead.

camilonova commented:
Awesome. I just updated it. How about now?

timgraham commented on tests/auth_tests/test_views.py:
```diff
@@ -336,7 +336,9 @@ def test_confirm_login_post_reset(self):
     )
     def test_confirm_login_post_reset_custom_backend(self):
-        backend = 'django.contrib.auth.backends.ModelBackend'
+        # Please note this backend is set at the url.
```
`# This backend is specified in the url().`

timgraham commented on tests/auth_tests/test_views.py:
```diff
@@ -336,7 +336,9 @@ def test_confirm_login_post_reset(self):
-        backend = 'django.contrib.auth.backends.ModelBackend'
+        # Please note this backend is set at the url.
+        backend = 'django.contrib.auth.backends.AllowAllUsersModelBackend'
+
```
skip the blank line

camilonova commented:
Looks like its ready
